src/Countries/BosniaAndHerzegovina.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class BosniaAndHerzegovina extends Country
+{
+    protected function __construct(
+        protected ?string $region = null,
+    ) {
+    }
+
+    public function countryCode(): string
+    {
+        return 'ba';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'Nova godina - prvi dan' => '01-01',
+            'Nova godina - drugi dan' => '01-02',
+            'Badnji dan (za pravoslavce)' => '01-06',
+            'Božić (za pravoslavce)' => '01-07',
+            'Praznik rada - prvi dan' => '05-01',
+            'Praznik rada - drugi dan' => '05-02',
+            'Badnji dan (za rimokatolike)' => '12-24',
+            'Božić (za rimokatolike)' => '12-25',
+        ], $this->regionHolidays());
+    }
+
+    /** @return array<string, string> */
+    protected function regionHolidays(): array
+    {
+        switch ($this->region) {
+            case 'BA-RS':
```

Comment:
I prefer to lowercase the incoming region variable and use lowercased cases

Reply:
@Nielsvanpach I take a look at some of the already implemented countries (Germany, to be more precise) and I saw they used uppercases there, so that's the reason why I decided to use uppercase